src/library/serato/seratofeature.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+// seratofeature.h
+// Created 2020-01-31 by Jan Holthuis
+//
+// This feature reads tracks and crates from removable Serato Libraries,
+// either in the Music directory or on removable devices (USB drives, etc),
+// by parsing the contents of the _Serato_ directory on each device.
+//
+// Most of the groundwork for this has been done here:
+//
+//      https://github.com/Holzhaus/serato-tags
+//      https://github.com/Holzhaus/serato-tags/blob/master/scripts/database_v2.py
+
+#include <QFuture>
+#include <QFutureWatcher>
+#include <QStringListModel>
+#include <QtConcurrentRun>
+#include <fstream>
+
+#include "library/baseexternallibraryfeature.h"
+#include "library/baseexternalplaylistmodel.h"
+#include "library/baseexternaltrackmodel.h"
+#include "library/treeitemmodel.h"
+
+class TrackCollectionManager;
+class BaseExternalPlaylistModel;
+
+class SeratoPlaylistModel : public BaseExternalPlaylistModel {
```

Comment:
Extracting this class into separate source files would help to shrink the huge .cpp file.

Reply:
good idea